Makefile

```diff
@@ -1,115 +1,98 @@
-	$(eval export LAUNCH_CONFIG_AWS)
-	@echo "$$LAUNCH_CONFIG_AWS" > $@
-    $(eval SSH_USER := centos)
-else ifeq ($(PLATFORM), azure)
-	$(eval export LAUNCH_CONFIG_AZURE)
-	@echo "$$LAUNCH_CONFIG_AZURE" > $@
-		$(eval SSH_USER := dcos)
-else
-	$(eval export LAUNCH_CONFIG_GCE)
-	@echo "$$LAUNCH_CONFIG_GCE" > $@
-endif
+define get_master_elb_ip
+$(shell test -f $(MASTER_ELB_IP_FILE) || \
+	terraform output -state=.deploy/terraform.tfstate "Master ELB Address" > $(MASTER_ELB_IP_FILE))
+$(eval MASTER_ELB_IP := $(shell cat $(MASTER_ELB_IP_FILE)))
+endef
 
+plan-dcos:
+	cd .deploy; \
+	terraform plan -var-file desired_cluster_profile
 
-destroy-dcos:
-	dcos-launch delete
-	$(RM) $(CLUSTER_INFO_FILE)
-	$(RM) $(LAUNCH_CONFIG_FILE)
-	$(RM) $(MASTER_IP_FILE)
-	$(RM) $(ID_FILE)
+launch-dcos:
+	cd .deploy; \
+	terraform apply -var-file desired_cluster_profile
 
-define rand_name
-$(shell cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 8 | head -n 1)
-endef
+kubectl-config:
+	kubectl config set-cluster dcos-k8s --server=http://localhost:9000
+	kubectl config set-context dcos-k8s --cluster=dcos-k8s --namespace=default
+	kubectl config use-context dcos-k8s
 
-kubectl-tunnel: $(ID_FILE)
+kubectl-tunnel:
 	$(call get_master_ip)
-	ssh -i $(ID_FILE) -4 -f -o "UserKnownHostsFile=/dev/null" -o "StrictHostKeyChecking=no" -o "ServerAliveInterval=120" \
-		 -N -L 9000:apiserver-insecure.kubernetes.l4lb.thisdcos.directory:9000 \
+	ssh -4 -o "UserKnownHostsFile=/dev/null" -o "StrictHostKeyChecking=no" -o "ServerAliveInterval=120" \
+		-N -L 9000:apiserver-insecure.kubernetes.l4lb.thisdcos.directory:9000 \
 		$(SSH_USER)@$(MASTER_IP)
 
-docker-build:
-	docker build -t mesosphere/dcos-kubernetes .
-
-GOOGLE_APPLICATION_CREDS := $(if ${GOOGLE_APPLICATION_CREDENTIALS},${GOOGLE_APPLICATION_CREDENTIALS},credentials.json)
-NUM_PRIVATE_AGENTS := $(if ${NUM_PRIVATE_AGENTS},${NUM_PRIVATE_AGENTS},3)
-NUM_PUBLIC_AGENTS := $(if ${NUM_PUBLIC_AGENTS},${NUM_PUBLIC_AGENTS},1)
-NUM_MASTERS := $(if ${NUM_MASTERS},${NUM_MASTERS},1)
-define docker_container
-	docker run -i \
-		-v $(GOOGLE_APPLICATION_CREDS):/credentials.json \
-		-e GCE_CREDENTIALS_PATH=/credentials.json \
-		-e AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY} \
-		-e AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID} \
-		-e NUM_PRIVATE_AGENTS=${NUM_PRIVATE_AGENTS} \
-		-e NUM_PUBLIC_AGENTS=${NUM_PUBLIC_AGENTS} \
-		-e NUM_MASTERS=${NUM_MASTERS} \
-		-e AZURE_CLIENT_ID=${AZURE_CLIENT_ID} \
-		-e AZURE_CLIENT_SECRET=${AZURE_CLIENT_SECRET} \
-		-e AZURE_SUBSCRIPTION_ID=${AZURE_SUBSCRIPTION_ID} \
-		-e AZURE_TENANT_ID=${AZURE_TENANT_ID} \
-		-v $(PWD):/dcos-kubernetes \
-		$(2) mesosphere/dcos-kubernetes $(1)
-endef
-
-docker:
-	$(call docker_container, /bin/bash, -t)
-
+plan: plan-dcos
 
 deploy: launch-dcos setup-cli install
+
+destroy-dcos:
+	$(RM) $(MASTER_IP_FILE)
+	$(RM) $(MASTER_ELB_IP_FILE)
+	cd .deploy; \
+	terraform destroy -var-file desired_cluster_profile
+
+clean:
+	$(RM) -r .deploy
+	$(RM) dcos
+	$(RM) kubectl
```

Comment:
What if `kubectl` was previously installed?

Reply:
I clean up the only current folder

Reply:
addressed not to remove kubectl